fiks-meldingsformidler/src/main/java/no/difi/meldingsutveksling/ks/svarinn/SvarInnService.java

```diff
@@ -1,133 +1,92 @@
-                          SvarInnFileDecryptor decryptor,
-                          SvarInnUnzipper unzipper,
-                          NoarkClient noarkClient,
-                          NoarkClient mailClient,
-                          ConversationService conversationService,
-                          IntegrasjonspunktProperties properties) {
-        this.svarInnClient = svarInnClient;
-        this.decryptor = decryptor;
-        this.unzipper = unzipper;
-        this.noarkClient = noarkClient;
-        this.mailClient = mailClient;
-        this.conversationService = conversationService;
-        this.properties = properties;
-        svarInnFileFactory = new SvarInnFileFactory();
+
+@RequiredArgsConstructor
+@ConditionalOnProperty(name = "difi.move.feature.enableDPF", havingValue = "true")
+@Component
+public class SvarInnService {
+
+    private final SvarInnClient svarInnClient;
+    private final SvarInnFileDecryptor decryptor;
+
+    public Stream<SvarInnStreamedFile> getAttachments(Forsendelse forsendelse) {
+        InputStream encrypted = svarInnClient.downloadZipFile(forsendelse);
+        InputStream decrypted = decryptor.decryptCMSStreamed(encrypted);
+        return unzip(forsendelse, decrypted);
     }
 
-    @Override
-    public void downloadFiles() {
-        final List<Forsendelse> forsendelses = svarInnClient.checkForNewMessages();
-        if (!forsendelses.isEmpty()) {
-            Audit.info(format("%d new messages in FIKS", forsendelses.size()));
-        }
-        for(Forsendelse forsendelse : forsendelses) {
-            Audit.info(format("Downloading message with fiks-id %s", forsendelse.getId()), Markers.append("fiks-id", forsendelse.getId()));
-            final SvarInnFile svarInnFile = svarInnClient.downloadFile(forsendelse.getDownloadUrl());
-            final byte[] decrypt = decryptor.decrypt(svarInnFile.getContents());
-            final Map<String, byte[]> unzippedFile;
-            try {
-                unzippedFile = unzipper.unzip(decrypt);
-            } catch (IOException e) {
-                throw new SvarInnForsendelseException("Unable to unzip file", e);
-            }
-            if (unzippedFile.values().isEmpty()) {
-                Audit.error("Zipfile is empty: skipping message", Markers.append("fiks-id", forsendelse.getId()));
-                continue;
-            }
-            // create SvarInnFile with unzipped file and correct mimetype
-            final List<SvarInnFile> files = svarInnFileFactory.createFiles(forsendelse.getFilmetadata(), unzippedFile);
+    private Stream<SvarInnStreamedFile> unzip(Forsendelse forsendelse, InputStream decrypted) {
+        Map<String, String> mimeTypeMap = forsendelse.getFilmetadata()
+                .stream()
+                .collect(Collectors.toMap(p -> p.get("filnavn"), p -> p.get("mimetype")));
+
+        Iterator<SvarInnStreamedFile> sourceIterator = new Iterator<SvarInnStreamedFile>() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 28 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1188)